### ISSUE TITLE
Optimize tree in panel for a product rule

### ISSUE THREAD

**Issue by bjarnef** [TeaCommerce/Tea-Commerce-for-Umbraco#7]
For discount codes I can also add a "Product Rule", however it looks a bit weird with the tree, tested in Umbraco 7.2 with the new starterkit "Fanoe" included.

![2014-12-06_06-49-40](https://cloud.githubusercontent.com/assets/2919859/5342703/4f32c238-7f00-11e4-8326-0b306f351bde.png)

I would expect it to look more like this:

![2014-12-06_06-53-22](https://cloud.githubusercontent.com/assets/2919859/5342717/6c69169a-7f00-11e4-8784-d80e049d0347.jpg)

**Comment by andersburla** [comment]
I think this is because Tea Commerce is loaded in a frame because of supporting umbraco v6. As long as Tea Commerce is loaded in a frame I dont think it will look 100% like Umbraco

**Comment by bjarnef** [comment]
@andersburla could you take a look at how it is done in https://github.com/kipusoep/UrlTracker when you click "Choose" for Redirect node?

The UrlTracker dashboard is loaded in an iframe too.

![2015-04-04_18-04-07](https://cloud.githubusercontent.com/assets/2919859/6993435/15f236b0-daf5-11e4-8a31-63f50ff2993f.jpg)

**Comment by bjarnef** [comment]
@andersburla does it make a difference if you use ```<umb:ContentPicker runat="server" ID="myID" />``` https://github.com/kipusoep/UrlTracker/blob/053c3ffdc15d90fcccff10d6505311053d4847f5/UI/UserControls/CreateView.ascx#L43 instead of this line? https://github.com/TeaCommerce/Tea-Commerce-for-Umbraco/blob/1e6a839f82dc10e090b3e946b89986d9946a0b17/Source/TeaCommerce.Umbraco.Application/Marketing/Rules/ProductRule/product-rule.html#L7

or what is different from core except ```prevent-default```?
https://github.com/umbraco/Umbraco-CMS/blob/ded1def8e2e7ea1a4fd0f849cc7a3f1f97cd8242/src/Umbraco.Web.UI.Client/src/views/prevalueeditors/treepicker.html#L10

or do the classes on the container ```class="umb-editor umb-contentpicker"``` have any effect?

**Comment by andersburla** [comment]
The umb:ContentPicker is a user control where the angular stuff is pure javascript that uses what Umbraco gives us access to. So don't know why the Umbraco angular doesn't give the right look and feel. Maybe you can dig deeper into the Umrbaco angular code and find the right method or options to use to get the right look. Would be nice

**Comment by bjarnef** [comment]
You're right.. I think I for a moment thought that ```<umb:ContentPicker />``` magically was handled as a directive in Umbraco 7.

It might be because Tea Commerce is inside "/framed/" in url unlike UrlTracker.
In v7 core it use treepicker in dialogservice https://github.com/umbraco/Umbraco-CMS/blob/ded1def8e2e7ea1a4fd0f849cc7a3f1f97cd8242/src/Umbraco.Web.UI.Client/src/views/prevalueeditors/treepicker.controller.js#L29 ... not sure if it is possible to use that with fallback for v6?

**Comment by rasmusfjord** [comment]
Another Issue that has risen with the Umbraco UI update is that you cant scroll in tree because its legacy.
